Sources/TuistKit/Commands/TuistCommand.swift

```diff
@@ -74,6 +73,24 @@ public struct TuistCommand: ParsableCommand {
         }
     }
 
+    private static func execute(_ command: ParsableCommand) throws {
+        var command = command
+        let userOptedOutFromStats = Environment.shared.tuistVariables["TUIST_STATS_OPT_OUT"] != nil
+        guard userOptedOutFromStats else {
+            try command.run()
+            return
+        }
+        let commandCompletionGroup = DispatchGroup()
+        commandCompletionGroup.enter()
+        let trackableCommand = TrackableCommand(command: command)
+        try trackableCommand.run {
+            commandCompletionGroup.leave()
+        }
+        // Block Tuist to wait until the event is persisted, otherwise it could get lost
+        // Note: Tuist is not waiting for the event to be successfully sent, but only persisted
+        _ = commandCompletionGroup.wait(timeout: DispatchTime.now() + DispatchTimeInterval.seconds(2))
```

Comment:
Are you assuming here that persisting == 2 seconds? If so I'd change the implementation to not make these assumptions.

Reply:
no, this is just a timeout in case `AsyncQueue` doesn't complete for some reasons, we exit the process

Reply:
Can we avoid time-dependent implementations?

Reply:
you need to have a backup timeout when you are waiting for an event to happen and your process might exit before that, right? Otherwise how do you make sure you don't wait forever if this event never gets fired? As far as I know most reactive implementations as well as iOS [expectation](https://developer.apple.com/documentation/xctest/asynchronous_tests_and_expectations/testing_asynchronous_operations_with_expectations) on XCTest do